esp-hal/src/gpio/mod.rs

```diff
@@ -11,28 +11,26 @@
 #![doc = ""]
 //! ## Working with pins
 //!
-//! Before use, the GPIO module must be initialized by creating an instance of
-//! the [`Io`] struct. This struct provides access to the pins on the chip.
-//!
-//! The [`Io`] struct can also be used to configure the interrupt handler for
-//! GPIO interrupts. For more information, see the
-//! [`Io::set_interrupt_handler`].
-//!
-//! The pins are accessible via the [`crate::Peripherals`] struct returned by
-//! [`crate::init`]. These pins can then be passed to peripherals (such as
-//! SPI, UART, I2C, etc.), to pin drivers or can be [`GpioPin::split`] into
-//! peripheral signals.
-//!
-//! Each pin is a different type initially. Internally, `esp-hal` will often
-//! erase their types automatically, but they can also be converted into
-//! [`AnyPin`] manually by calling [`Pin::degrade`].
+//! After initializing the HAL, you can access the individual pins using the
+//! [`crate::Peripherals`] struct. These pins can then be used as general
+//! purpose digital IO using pin drivers, or the can be passed to peripherals
+//! (such as SPI, UART, I2C, etc.), or can be [`GpioPin::split`]
+//! into peripheral signals for advanced use.
 //!
 //! Pin drivers can be created using [`Flex::new`], [`Input::new`],
 //! [`Output::new`] and [`OutputOpenDrain::new`]. If you need the pin drivers to
 //! carry the type of the pin, you can use the [`Flex::new_typed`],
 //! [`Input::new_typed`], [`Output::new_typed`], and
 //! [`OutputOpenDrain::new_typed`] functions.
 //!
+//! Each pin is a different type initially. Internally, `esp-hal` will often
+//! erase their types automatically, but they can also be converted into
+//! [`AnyPin`] manually by calling [`Pin::degrade`].
+//!
+//! The [`Io`] struct can also be used to configure the interrupt handler for
+//! GPIO interrupts. For more information, see the
+//! [`Io::set_interrupt_handler`].
```

Comment:
`set_interrupt_handler` deserves an example, but I think on the method, not on the module.

Reply:
At the very least we have an example usage of this in the examples folder for GPIO, which is a bit of weirder case.